core/src/main/java/io/questdb/cairo/AbstractIndexReader.java

```diff
@@ -50,10 +50,10 @@ public abstract class AbstractIndexReader implements BitmapIndexReader {
 
     @Override
     public void close() {
-        if (isOpen()) {
-            Misc.free(keyMem);
-            Misc.free(valueMem);
-        }
+        // Do not add isOpen() condition here.
+        // Failures in opening should have memories closed as well.
```

Comment:
isOpen also prevented double-close issues. It would be more prudent to set memory variables to null if they are freed, e.g. `keyMem = Misc.free(keyMem)`

Reply:
We close memory objects even when they are not open or even half way though opening, memory classes handle that internally. There is no way to produce double close by that.